src/chainlit/frontend/src/App.tsx

```diff
@@ -13,21 +13,28 @@ import SettingsModal from 'components/molecules/settingsModal';
 import Socket from 'components/socket';
 
 import { useAuth } from 'hooks/auth';
+import { useApi } from 'hooks/useApi';
 
-import { clientState } from 'state/client';
 import { settingsState } from 'state/settings';
 import { accessTokenState, roleState } from 'state/user';
 
 import './App.css';
 
 function App() {
-  const client = useRecoilValue(clientState);
   const { theme: themeVariant } = useRecoilValue(settingsState);
   const [accessToken, setAccessToken] = useRecoilState(accessTokenState);
-  const setRole = useSetRecoilState(roleState);
+  const [role, setRole] = useRecoilState(roleState);
   const { isAuthenticated, getAccessTokenSilently, logout } = useAuth();
   const theme = makeTheme(themeVariant);
 
+  const { data: roleData, error: roleError } = useApi(
```

Comment:
the previous version was waiting to receive the access token to fetch the role. Will this version execute before having the access token and a second time after receiving it?